### PR TITLE
Log partition movements before throwing exception

### PR DESCRIPTION
**Pull request by mhratson**
## Details

- Provide partition movement details in`OngoingExecutionException` exception message.
- Having partition movement details is handy for troubleshooting in cases when partition movements are ongoing for a long time, providing necessary details for potential mitigation.

### CODE REVIEW

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/Executor.java [hunk 1]
```diff
@@ -1048,15 +1049,19 @@ private void startExecution(LoadMonitor loadMonitor,
    */
   private void sanityCheckOngoingMovement() throws OngoingExecutionException {
     boolean hasOngoingPartitionReassignments;
+    Map<TopicPartition, PartitionReassignment> ongoingPartitionReassignments;
     try {
-      hasOngoingPartitionReassignments = hasOngoingPartitionReassignments();
+       ongoingPartitionReassignments = ExecutionUtils.ongoingPartitionReassignments(_adminClient);
+      // avoid calling hasOngoingPartitionReassignments() to save 1 request.
```
Comment (CCisGG) [inline cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/Executor.java]: I think it's fine to remove this comment. This is probably only be useful during review time.

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/Executor.java [hunk 2]
```diff
@@ -1048,15 +1049,19 @@ private void startExecution(LoadMonitor loadMonitor,
...
+      hasOngoingPartitionReassignments = !ongoingPartitionReassignments.keySet().isEmpty();
     } catch (TimeoutException | InterruptedException | ExecutionException e) {
       // This may indicate transient (e.g. network) issues.
       throw new IllegalStateException("Failed to retrieve if there are already ongoing partition reassignments.", e);
     }
     // Note that in case there is an ongoing partition reassignment, we do not unpause metric sampling.
     if (hasOngoingPartitionReassignments) {
-      throw new OngoingExecutionException("There are ongoing inter-broker partition movements.");
+      LOG.error("Ongoing inter-broker partition movements: {}", ongoingPartitionReassignments);
+      throw new OngoingExecutionException("There are ongoing inter-broker partition movements");
```
Comment (CCisGG) [inline cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/Executor.java]: I think the error log here could be redundant. Ideally you only need this:

`throw new OngoingExecutionException("There are ongoing inter-broker partition movements: " + ongoingPartitionReassignments); `

And then the caller whoever handle that exception should be able to log error.

**Review by CCisGG — CHANGES_REQUESTED** [review]

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/Executor.java (hunk 2 above)
Comment (mhratson) [reply cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/Executor.java]: My concern is if there's alot of partition movements the exception message will be pretty big which is not exactly desirable output. 
But if that's acceptable then i'm cool with that. 
Will make updates.

Comment (CCisGG) [reply cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/Executor.java]: That should be fine. It will be logged either ways.

**Review by CCisGG — APPROVED** [review]

**Pull request closed without merging**